### PR TITLE
Adds h2c support

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Enables HTTP/2 connections over cleartext TCP with Prior Knowledge (RFC 7540 3.4).

The implementation is based on the golang.org/x/net/http2/h2c and workarounds several issues:
* https://github.com/golang/go/issues/38064
* ~https://github.com/golang/go/issues/26682~ https://github.com/zalando/skipper/pull/1868#pullrequestreview-1525932331

See h2c package docs for details.

Signed-off-by: Alexander Yastrebov <alexander.yastrebov@zalando.de>

### CODE REVIEW

**Comment by tsingson** [comment]
> From my side great first step to support h2c as a server handler!

any update about h2c ?

**Comment by szuecs** [comment]
@tsingson I think we should wait for https://github.com/golang/net/pull/139 being merged and released, if not it's not reasonable to have h2c support in Go.

**Comment by tsingson** [comment]
> @tsingson I think we should wait for [golang/net#139](https://github.com/golang/net/pull/139) being merged and released, if not it's not reasonable to have h2c support in Go.

thanks for update.

**Comment by tsingson** [comment]
https://go-review.googlesource.com/c/net/+/419181

good news.

File: h2c/h2c.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+		if n == 0 {
+			return serr
+		}
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-timer.C:
+		}
+	}
+}
```
Comment (AlexanderYastrebov) [inline h2c/h2c.go:149]: This is just waiting for connections close. Proper shutdown should send goaway
* https://github.com/golang/net/blob/304cc91b19ae873219f3d0807c8533267629cf2e/http2/server.go#L232-L241
* https://github.com/golang/net/blob/304cc91b19ae873219f3d0807c8533267629cf2e/http2/server.go#L1384-L1394

(i.e. https://github.com/golang/go/issues/26682 is not solved)

**Comment by AlexanderYastrebov** [comment]
https://github.com/golang/net/pull/139 propagates http.Server instance and therefore vrious timeouts. The problem of graceful shutdown is not solved yet.

**Comment by AlexanderYastrebov** [comment]
Closing in favour of #2480

**Pull request closed without merging**